scripts/obico.sh

```diff
@@ -0,0 +1,422 @@
+  echo "${services}"
+}
+
+function cfg_dir() {
+  local name=${1}
+  if [[ -z ${name} || ${name} == "moonraker" || ${name} == "moonraker-obico" ]]; then
+    echo "${KLIPPER_CONFIG}"
+  else
+    local re="^[1-9][0-9]*$"
+    ### overwrite config folder if name is only a number
+    if [[ ${name} =~ ${re} ]]; then
+      echo "${KLIPPER_CONFIG}/printer_${name}"
+    else
+      echo "${KLIPPER_CONFIG}/${name}"
+    fi
+  fi
+}
+
+function is_moonraker_obico_linked() {
+  local name=${1}
+  moonraker_obico_cfg="$(cfg_dir ${name})/moonraker-obico.cfg"
+  grep -s -E "^[^#]" "${moonraker_obico_cfg}" | grep -q 'auth_token'
+  return $?
+}
+
+function get_moonraker_names() {
+  local moonraker_services
+  moonraker_services=$(moonraker_systemd)
+  if [[ -z ${moonraker_services} ]]; then
+    echo '' && return
+  fi
+
+  for service in ${moonraker_services}; do
+    get_instance_name "${service}" moonraker
+  done
+}
+
+function obico_server_url_prompt() {
+  top_border
+  printf "|${green}%-55s${white}|\n" " Obico Server URL"
+  blank_line
+  echo -e "| You can use a self-hosted Obico Server or the Obico   |"
+  echo -e "| Cloud. For more information, please visit:            |"
+  echo -e "| https://obico.io.                                     |"
+  blank_line
+  echo -e "| For the Obico Cloud, leave it as the default:         |"
+  printf "|${cyan}%-55s${white}|\n" " https://app.obico.io"
+  blank_line
+  echo -e "| For self-hosted server, specify:                      |"
+  printf "|${cyan}%-55s${white}|\n" " http://server_ip:port"
+  echo -e "| For instance, \`http://192.168.0.5:3334\`.              |"
+  bottom_border
+}
+
+function moonraker_obico_setup_dialog() {
+  status_msg "Initializing Moonraker-obico installation ..."
+
+  ### return early if moonraker is not installed
+  local moonraker_count moonraker_names
+  moonraker_names=($(get_moonraker_names))
+  moonraker_count=${#moonraker_names[@]}
+  if (( moonraker_count == 0 )); then
+    local error="Moonraker not installed! Please install Moonraker first!"
+    log_error "Moonraker-obico setup started without Moonraker being installed. Aborting setup."
+    print_error "${error}" && return
+  fi
+
+  local moonraker_obico_services moonraker_obico_names=()
+  moonraker_obico_services=$(moonraker_obico_systemd)
+  existing_moonraker_obico_count=$(echo "${moonraker_obico_services}" | wc -w )
+  for service in ${moonraker_obico_services}; do
+    moonraker_obico_names+=( "$(get_instance_name "${service}" moonraker-obico)" )
+  done
+
+  local allowed_moonraker_obico_count=$(( moonraker_count - existing_moonraker_obico_count ))
+  if (( allowed_moonraker_obico_count > 0 )); then
+
+    ### Step 1: Ask for the number of moonraker-obico instances to install
+    if (( moonraker_count == 1 )); then
+      ok_msg "Moonraker installation found!\n"
+      new_moonraker_obico_count=1
```

Comment:
```ssh 
new_moonraker_obico_count=1
```
I'm unsure about the usage of this variable. It seems to be not used anywhere? Is it used somewhere else as it is not declared as a local variable?

Reply:
Good catch. It's a bug! Do you want me to push this change?

```
diff --git a/scripts/obico.sh b/scripts/obico.sh
index 791fcdb..de2bd3f 100644
--- a/scripts/obico.sh
+++ b/scripts/obico.sh
@@ -98,7 +98,6 @@ function moonraker_obico_setup_dialog() {
     ### Step 1: Ask for the number of moonraker-obico instances to install
     if (( moonraker_count == 1 )); then
       ok_msg "Moonraker installation found!\n"
-      new_moonraker_obico_count=1
     elif (( moonraker_count > 1 )); then
       top_border
       printf "|${green}%-55s${white}|\n" " ${moonraker_count} Moonraker instances found!"
@@ -134,7 +133,7 @@ function moonraker_obico_setup_dialog() {
         (( new_moonraker_obico_count > allowed_moonraker_obico_count )) && error_msg "Number of Moonraker-obico instances larger than installed Moonraker instances"
       done && select_msg "${new_moonraker_obico_count}"
     else
-      log_error "Internal error. new_moonraker_obico_count of '${new_moonraker_obico_count}' not equal or grather than one!"
+      log_error "Internal error. moonraker_count of '${moonraker_count}' not equal or grather than one!"
       return 1
     fi
```